app/src/main/java/space/taran/arkretouch/presentation/edit/rotate/RotateOperation.kt

```diff
@@ -13,4 +20,26 @@ class RotateOperation(private val editManager: EditManager) : Operation {
             addRotation()
         }
     }
+
+    override fun undo() {
+        editManager.apply {
+            if (rotationAngles.isNotEmpty()) {
+                redoRotationAngles.push(prevRotationAngle)
+                prevRotationAngle = rotationAngles.pop()
+                matrix.reset()
+                rotate(prevRotationAngle)
+            }
+        }
+    }
+
+    override fun redo() {}
```

Comment:
Not implemented yet.

Reply:
Do we really need `redo`? Can we just call `apply` when we need redo?

Reply:
Yes, we need it for now because ```apply``` and ```redo``` have different logic. But we have to make the logic the same later, if possible, and then we can omit ```redo```